### PR TITLE
[ONEM-16360] Parameter Separator support

### PR DESCRIPTION
**Pull request by erenatas**
Signed-off-by: Eren ATAS <eatas.contractor@libertyglobal.com>

### CODE REVIEW

File: jenkins_jobs/modules/parameters.py [hunk 1]
```diff
@@ -1112,6 +1112,36 @@ def git_parameter_param(registry, xml_parent, data):
     ]
     helpers.convert_mapping_to_xml(pdef, data, mapping, fail_required=True)
 
+def param_separator(registry, xml_parent, data):
+    """yaml: parameter-separator
+    A parameter separator.
+
+    :arg str name: UUID of the separator (optional)
```
Comment (romanek-adam) [inline jenkins_jobs/modules/parameters.py]: I'd describe it as: name of the separator (optional, the plugin will assign a randomly generated UUID if not specified)

File: jenkins_jobs/modules/parameters.py [hunk 2]
```diff
@@ -1112,6 +1112,36 @@ def git_parameter_param(registry, xml_parent, data):
...
+    :arg str separator-style: the style of the separator (optional)
```
Comment (romanek-adam) [inline jenkins_jobs/modules/parameters.py]: It's worth to mention this is CSS, just as for `section-header-style`

File: jenkins_jobs/modules/parameters.py [hunk 3]
```diff
@@ -1112,6 +1112,36 @@ def git_parameter_param(registry, xml_parent, data):
...
+        - parameter-separator:
+            name: lorem
+            separator-style: FOO
+            section-header: bar
+            section-header-style: font-weight:bold;z-index:10000
+    """
+    element_name = (
+        "jenkins.plugins.parameter__separator.ParameterSeparatorDefinition"
+    )
+    pdef = XML.SubElement(xml_parent, element_name)
```
Comment (romanek-adam) [inline jenkins_jobs/modules/parameters.py:1135]: Don't we need an empty `<description/>` tag, even though it's not used?

Comment (erenatas) [reply jenkins_jobs/modules/parameters.py:1135]: We don't need it, it does not have `description` tag.

**Comment by romanek-adam** [comment]
@erenatas please run `tox` to reformat the code properly.

File: jenkins_jobs/modules/parameters.py (hunk 3 above)
Comment (romanek-adam) [reply jenkins_jobs/modules/parameters.py:1135]: Well the old XML for binpkg does, right?

**Review by romanek-adam — APPROVED** [review]

**Review by vdanaila — APPROVED** [review]

**Comment by romanek-adam** [comment]
@erenatas please upstream if not done yet.

**Pull request merged**